src/main/java/net/sf/jsqlparser/Model.java

```diff
@@ -0,0 +1,14 @@
+/*-
+ * #%L
+ * JSQLParser library
+ * %%
+ * Copyright (C) 2004 - 2020 JSQLParser
+ * %%
+ * Dual licensed under GNU LGPL 2.1 or Apache License 2.0
+ * #L%
+ */
+package net.sf.jsqlparser;
+
+public interface Model {
```

Comment:
Lets add a bit of doc that this is a marker interface for anything that is a certain type (in order to be able to make a limiting parameter that isn't Object). (Or something of that nature), just seeing a Model type with no methods might not explain our intent with what a `Model` is.

Reply:
Maybe we should rename it to SqlModel?

Reply:
https://github.com/JSQLParser/JSqlParser/pull/1014/commits/9a9422fa4b0459b62623f884032176e6872e2cf7

Reply:
SqlModel would be a good name. That doc is good tho!

Reply:
Don't know if we need this kind of marker, but it's sometimes helpful to group things into a type hierarchy, which shares a common usage or programming style (with-/add methods, casting getters) to help others on implementation new model types

Reply:
Yeah, a marker works :)